display_books_promise.js

```diff
@@ -0,0 +1,65 @@
+let BookData = new Array();
+fetch("random_books.json")
+  .then((response) => response.json())
+  .then((data) => {
+    BookData = data;
+    insertDataInHtml(BookData);
+  })
+  .catch((error) => console.log(error));
+
+document.getElementById("btn-search-id").onclick = IdButton;
+document.getElementById("btn-search-genre").onclick = genreButton;
+document.getElementById("btn-search-price").onclick = priceButton;
+
+function IdButton(event) {
+  event.preventDefault();
+  const idValue = document.getElementById("input-search-id").value;
+  if (idValue.trim().length <= 0) return;
```

Comment:
I am not sure why length will go less than 0. 
The above line can be written as - 
```
if (i!dValue.trim().length) return;
```

Please validate.

Reply:
Resolved